src/ParallelAlgorithms/Amr/Actions/EvaluateRefinementCriteria.hpp

```diff
@@ -97,6 +100,17 @@ struct EvaluateRefinementCriteria {
       }
     }
 
+    // An element cannot join if it is splitting in another dimension
+    if (alg::any_of(overall_decision, [](amr::domain::Flag flag) {
+          return flag == amr::domain::Flag::Split;
+        })) {
+      for (size_t d = 0; d < volume_dim; ++d) {
+        if (gsl::at(overall_decision, d) == amr::domain::Flag::Join) {
+          gsl::at(overall_decision, d) = amr::domain::Flag::DoNothing;
```

Comment:
Does this logic have to be repeated here? The AMR logic is complicated enough, so it would be quite helpful to confine it to as few places as possible. If you put this here so it runs even when the element has no neighbors, then I suggest to allow the update function to work for the no-neighbors case.

Reply:
Yes, as I think it is better to make this change before sending the flags to neighbors, as each time `update_amr_decision` changes a flag it triggers sending the updated flags to each neighbor...

Reply:
maybe factor out the logic into a separate function, and add some comments explaining what you wrote here?